packages/shared-ux/chrome/navigation/__jest__/build_nav_tree.test.tsx

```diff
@@ -0,0 +1,738 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import './setup_jest_mocks';
+import React from 'react';
+import { type RenderResult } from '@testing-library/react';
+import { type Observable, of } from 'rxjs';
+import type { ChromeNavLink } from '@kbn/core-chrome-browser';
+
+import { navLinksMock } from '../mocks/src/navlinks';
+import { Navigation } from '../src/ui/components/navigation';
+import type { RootNavigationItemDefinition } from '../src/ui/types';
+
+import {
+  getMockFn,
+  renderNavigation,
+  errorHandler,
+  type TestType,
+  type ProjectNavigationChangeListener,
+} from './utils';
+
+describe('builds navigation tree', () => {
+  test('render reference UI and build the navigation tree', async () => {
+    const onProjectNavigationChange = getMockFn<ProjectNavigationChangeListener>();
+
+    const runTests = async (type: TestType, { findByTestId }: RenderResult) => {
+      try {
+        expect(await findByTestId(/nav-item-group1.item1\s/)).toBeVisible();
+        expect(await findByTestId(/nav-item-group1.item2\s/)).toBeVisible();
+        expect(await findByTestId(/nav-item-group1.group1A\s/)).toBeVisible();
+        expect(await findByTestId(/nav-item-group1.group1A.item1\s/)).toBeVisible();
+        expect(await findByTestId(/nav-item-group1.group1A.group1A_1\s/)).toBeVisible();
+
+        // Click the last group to expand and show the last depth
+        (await findByTestId(/nav-item-group1.group1A.group1A_1\s/)).click();
+
+        expect(await findByTestId(/nav-item-group1.group1A.group1A_1.item1/)).toBeVisible();
+
+        expect(onProjectNavigationChange).toHaveBeenCalled();
+        const lastCall =
+          onProjectNavigationChange.mock.calls[onProjectNavigationChange.mock.calls.length - 1];
+        const [{ navigationTree }] = lastCall;
+        return navigationTree;
+      } catch (e) {
+        errorHandler(type)(e);
+      }
+    };
+
+    // -- Default navigation
+    {
```

Comment:
Interesting use of block scopes! Is it to work around variable names, like `conts renderResult`?

Reply:
Yep. I used to use `let` to be able to override later in my tests. I prefer this syntax (saw that in `core` codebase).